.github/workflows/typescript_tests.yml

```
@@ -145,7 +166,10 @@ jobs:
       - name: Install node
         uses: actions/setup-node@v4
         with:
-          node-version: '16'
+          node-version: '18'
+      - name: Install Develocity Agent
+        run: npm install https://develocity-npm-pkgs.gradle.com/gradle-develocity-agent-1.0.2.tgz
+        working-directory: ./sdks/typescript
```

Same comments as above apply here. I'll copy and paste them for your convenience.

---

```suggestion
        run: npm install -D @gradle-tech/develocity-agent@1.0.2
        working-directory: ./sdks/typescript
```
1. `-D` (alias of `--save-dev`) otherwise the agent is installed as a dependency of the package itself, rather than a `devDependency`.
1. adding the version number to prevent this from being upgraded to 2.0.0 before the servers are updated to 2025.3
1. pointing to the version on npmjs.com rather than our S3 bucket

---

```suggestion
        run: npm exec -y -- pacote extract @gradle-tech/develocity-agent ~/.node_libraries/@gradle-tech/develocity-agent@1.0.2
```
Even better: see if you can replace this with the new way we are recommending to install the plugin as a global package as documented [here](https://docs.grdev.net/develocity/npm-agent/#global_installation_well_known_folder) (pointing to grdev.net as this is still pre-release) -- installing as a global package is done to enable capturing dependencies (which you would do anyway here, since you're doing this as an earlier step, but I think it would be nice to test the way we are recommending users to do this). It also allows users to refer to our helpers regardless of where they put the configuration file. 🙂